src/test/java/at/ac/tuwien/kr/alpha/grounder/transformation/impl/StratifiedEvaluationRegressionTest.java

```diff
@@ -0,0 +1,255 @@
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("p", "a")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("q", "b", "c")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("r", "c", "c", "a")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("s", "b")));
+	}
+
+	private static void verifyAnswerSetsFactsOnly(Set<AnswerSet> answerSets) {
+		TestUtils.assertAnswerSetsEqual("a, b, c, p(a), q(b,c), r(c,c,a), s(b)", answerSets);
+	}
+
+	private static void verifyProgramStratNoFacts(InternalProgram evaluated) {
+		Assert.assertTrue(evaluated.getFacts().isEmpty());
+	}
+	
+	private static void verifyAnswerSetsStratNoFacts(Set<AnswerSet> answerSets) {
+		Assert.assertEquals(1, answerSets.size());
+		Assert.assertTrue(answerSets.iterator().next().isEmpty());
+	}
+	
+	private static void verifyProgramStratWithFacts(InternalProgram evaluated) {
+		// rules should all be taken care of at this point
+		Assert.assertTrue(evaluated.getRules().isEmpty());
+		Assert.assertTrue(evaluated.getRulesById().isEmpty());
+		Assert.assertTrue(evaluated.getPredicateDefiningRules().isEmpty());
+		
+		// facts should be the full answer set
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("req", "a")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("req", "b")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("incomp", "b")));
+		
+		// below facts from stratified evaluation
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("base", "a")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("depend_base", "a", "a")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("dep_b_hlp", "a")));
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("depend_further", "a")));
+	}
+	
+	private static void verifyAnswerSetsStratWithFacts(Set<AnswerSet> answerSets) {
+		TestUtils.assertAnswerSetsEqual("req(a), req(b), incomp(b), base(a), depend_base(a,a), dep_b_hlp(a), depend_further(a)", answerSets);
+	}
+	
+	private static void verifyProgramEquality(InternalProgram evaluated) {
+		Assert.assertEquals(0, evaluated.getRules().size());
+		Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("equal")));
+	}
+	
+	private static void verifyAnswerSetsEquality(Set<AnswerSet> answerSets) {
+		TestUtils.assertAnswerSetsEqual("equal", answerSets);
+	}
+	
+	private static void verifyProgramEqualityWithVar(InternalProgram evaluated) {
+		Assert.assertEquals(0, evaluated.getRules().size());
+		// TODO convenience method for atoms with numeric terms
+		//Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("b", "1")));
+		//Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("c", "2")));
+		//Assert.assertTrue(evaluated.getFacts().contains(BasicAtom.newInstance("d", "3")));
```

Comment:
?

Reply:
Fixed! The idea of the TODO was to add a simple factory method to ```BasicAtom``` to create atoms with integer arguments. Taken care of now..